PUBLISHING.md

```diff
@@ -59,6 +59,22 @@ Then `cd elm-tooling`, push the resulting branch, and make a PR to <https://gith
 
 ## NPM
 
+Repeat this for all the binary packages in `package/npm/packages/`. This uses `package/npm/packages/elm-format-darwin-x64` as an example.
+
+1. Go to the folder: `cd package/npm/packages/elm-format-darwin-x64`
+1. Copy the appropriate binary to `./elm-format`. For Windows: `./elm-format.exe`
+1. Double-check that you put the right binary in the right package: `file elm-format`
+1. Double-check that the file is executable: `ls -l elm-format`
+1. In `package.json` of the binary package, bump the version for example to `"0.8.6"`.
+1. In `package.json` of the main npm package, update `"optionalDependencies"` to point to the bumped version. For example: `"@avh4/elm-format-darwin-x64": "0.8.6"`.
+
+   Note: Pin the versions of the binary packages _exactly_ – no version ranges. This means that installing `elm-format@0.8.6` installs the exact same bytes in two years as today.
+1. Publish the package: `npm publish --access=public`
+
+   `--access=public` is needed because scoped packages are private by default.
```

Comment:
And just to double-check that I'm understanding how this works, the binary will actually be published into the npmjs repo (since there's no explicit manifest given, all the files in the local directory get published?), so it's now actually npm that's downloading the binary?

Reply:
Yes, the binary will end up in the tarball for the package and be stored in the npmjs registry, and npm will download the package as any other. An npm package is just a folder with a valid package.json and random files that npm doesn’t really care about.

Here’s an example: https://unpkg.com/browse/@lydell/elm_darwin_arm64@0.19.1-2/

If you `cd` into a folder with a `package.json` and run `npm publish`, here are some rules:

1. The contents of that folder is packed into a tarball and uploaded to the npm registry.
2. You can use an `.npmignore` file to exclude files, `.gitignore`-style.
3. You can use the `"files"` field in package.json to instead _only_ include the stuff listed there.
4. package.json is always included no matter what you do.
5. Some other files are always included as well, if they exist, such as README.md and LICENSE.

https://docs.npmjs.com/cli/v9/configuring-npm/package-json#files

I didn’t include the `"files"` field in the sub packages because:

1. I based much of the approach on esbuild, and they don’t have it: https://github.com/evanw/esbuild/blob/e345b13687bc3ac86f18f4a266a162653544ad31/npm/%40esbuild/darwin-arm64/package.json
2. It felt unlikely that there will be other random files in those folders.
3. It saves some bytes? 🙃 

But you could add `"files": ["elm-format"]` (`"files": ["elm-format.exe"]` for Windows) if you want, it doesn’t hurt. (package.json and README.md are automatically included always.)

Reply:
Awesome, thanks for the extra details and the unpkg.com/browse trick!